apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js

```diff
@@ -17,6 +17,63 @@ import tinymce from 'tinymce/tinymce';
  */
 import { inIframe, sendMessage } from './utils';
 
+/**
+ * Monitors Gutenburg for when an editor is opened with content originally authored in the classic editor.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function triggerConversionPrompt( calypsoPort ) {
+	function addConversionListener( classicBlock ) {
+		// Listen for user response
+		calypsoPort.addEventListener( 'message', onConversionResponse, false );
```

Comment:
We started recently to use specific ports rather than the _global_ `calypsoPort`. It'd be great to continue using that approach here. See this example ([#](https://github.com/Automattic/wp-calypso/blob/409f2815a5833349211d9275010362d20a913bab/apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js#L584-L606), [#](https://github.com/Automattic/wp-calypso/blob/409f2815a5833349211d9275010362d20a913bab/client/gutenberg/editor/calypsoify-iframe.tsx#L178), [#](https://github.com/Automattic/wp-calypso/blob/409f2815a5833349211d9275010362d20a913bab/client/gutenberg/editor/calypsoify-iframe.tsx#L279)) with the media modal. 

The idea is to sent a new `MessageChannel` port with the first message sent to Calypso. Then, Calypso will use that port to sent a message back while the server keeps listening to messages received on that port.